src/components/Dialog/index.test.js

```diff
@@ -31,9 +30,9 @@ describe('Dialog', () => {
 
     /* Tests that onDismiss gets fired correctly */
     /* ========================================= */
-    fireEvent.click(screen.getByLabelText('Close'));
+    userEvent.click(screen.getByLabelText('Close'));
     // Pressing esc anywhere in dialog should trigger a close
-    fireEvent.keyDown(screen.getByText('content'), { key: 'Escape', code: 27 });
+    userEvent.type(screen.getByText('content'), '{esc}');
```

Comment:
it may be the case that `fireEvent` should never be used alongside `userEvent` as it encompasses all the methods of the other.